core/core_bind.cpp

```diff
@@ -1508,7 +1508,7 @@ Variant _File::get_var(bool p_allow_objects) const {
 	ERR_FAIL_COND_V_MSG(!f, Variant(), "File must be opened before use.");
 	uint32_t len = get_32();
 	Vector<uint8_t> buff = get_buffer(len);
-	ERR_FAIL_COND_V((uint32_t)buff.size() != len, Variant());
+	ERR_FAIL_COND_V((uint32_t)buff.size() != len - 1, Variant());
```

Comment:
I'm against merging this.

As I explained in https://github.com/godotengine/godot/issues/47971#issuecomment-932963030 the regression bug happens with `open_compressed` but not with `open`. If we add this change, then it works when we use `open_compressed` but fails when we use `open`.

I would also like to point out that the same pattern is currently in other pars of master, These are a couple examples:

- https://github.com/godotengine/godot/blob/4633f8c3ba8fe16405f9ea03544cb0004f2216bc/core/io/resource_uid.cpp#L184
- https://github.com/godotengine/godot/blob/acc776f7b6ca43a6dde59f7355f52a11b9ac72f5/modules/gdnative/pluginscript/pluginscript_script.cpp#L450

You can find more if you search for it.